x/perpetual/keeper/open.go

```diff
@@ -45,6 +45,9 @@ func (k Keeper) Open(ctx sdk.Context, msg *types.MsgOpen, isBroker bool) (*types
 	if err != nil {
 		return nil, err
 	}
+	if !ammPool.PoolParams.UseOracle {
+		return nil, types.ErrPoolHasToBeOracle
+	}
```

Comment:
I think we should add it when updating or adding new pool too, right ?

Reply:
Hello Amit. We should create a new task for this to have a scope about the task